### PR TITLE
Template root path notifier (required for xml_plugin)

### PR DESCRIPTION
**Pull request by jan-vcapgemini**
Adresses #173.

Implements

* added new MERGE_SCHEMA_RESOURCE_FOLDER constant to api
* added new setProjectRoot method to GeneratorPluginActivator interface (used to update template root location on plugins)
* made sure to send an update request to all plugins when the path to the template root was changed

@devonfw/cobigen

### CODE REVIEW

File: cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/extension/PluginRegistry.java [hunk 1]
```diff
@@ -33,6 +37,11 @@
     private static Map<String, TriggerInterpreter> registeredTriggerInterpreter =
         Collections.synchronizedMap(Maps.<String, TriggerInterpreter> newHashMap());
 
+    /**
+     * List of registered plugins
+     */
+    private static List<Object> pluginsList = Lists.newLinkedList();
```
Comment (maybeec) [inline cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/extension/PluginRegistry.java:43]: better List<GeneratorPluginActivator>

File: cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/extension/PluginRegistry.java [hunk 2]
```diff
@@ -154,4 +165,23 @@ public static TriggerInterpreter getTriggerInterpreter(String triggerType) {
...
+    /**
+     * Notifies plugins about the new template root path
+     *
+     * @param configFolder
+     *            Path to update on registered plugins
+     */
+    public static void notifyPlugins(Path configFolder) {
+
+        for (Object plugin : pluginsList) {
+            if (plugin instanceof GeneratorPluginActivator) {
```
Comment (maybeec) [inline cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/extension/PluginRegistry.java:177]: check not needed anymore

File: cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/extension/PluginRegistry.java (hunk 1 above)
Comment (jan-vcapgemini) [reply cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/extension/PluginRegistry.java]: adjusted to native ArrayList.

File: cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/extension/PluginRegistry.java (hunk 2 above)
Comment (jan-vcapgemini) [reply cobigen/cobigen-core-parent/cobigen-core/src/main/java/com/devonfw/cobigen/impl/extension/PluginRegistry.java]: removed.

**Pull request merged**